vmaas/cache.go

```diff
@@ -70,6 +70,10 @@ type Cache struct {
 	OvalTestID2States               map[TestID][]OvalState
 	OvalDefinitionID2ErrataIDs      map[DefinitionID][]ErratumID
 	CpeID2Label                     map[CpeID]string
+
+	// CSAF
+	CSAFProductStatus map[int]string
+	CSAFCVEs          map[CSAFProduct][]CVEID
```

Comment:
I don't know what was discussed previously with Jan. To me it looks like we need also information about the product status, or do we want to infer it from the presence of package_name_id and package_id? I'm not saying this is wrong, I'm just trying to understand how we will know if the CVE is fixed or unfixed.

Reply:
Could we have a problem here because CSAFProduct struct contains pointers?

Let's have a CSAFProduct with PackageNameID=1, pointer of PackageNameID will be stored to CSAFProduct during the cache load and CVEs for given product will be added to map, so far so good.
Then we will try to use `CSAFCVEs` to get list of CVEs for a given product in evaluation. So we will create a CSAFProduct struct also with PackageNameID=1 but that would be a different instance of a variable containing value 1, it will have a different pointer. Therefore, such struct created in the evaluation won't be found in `CSAFCVEs` map.

I tried to replicate such scenario in yaegi
```
> x := 1
: 1
> y := 1
: 1
> xstruct := StructWPtr{ID: &x}
: {0x1400003c8f8}
> ystruct := StructWPtr{ID: &y}
: {0x14000212138}
> xstruct == ystruct
: false
> z := map[StructWPtr]bool{}
: map[]
> z[xstruct] = true
: false
> z[xstruct]
: true
> z[ystruct]
: false
```
I might be missing something or assuming something which isn't true, what do you think?